imap/mailbox.h

```diff
@@ -57,8 +57,9 @@
 #include "seqset.h"
 #include "util.h"
 
-#define MAX_MAILBOX_NAME 490
+#define MAX_MAILBOX_CREATENAME 490
 /* enough space for all possible rewrites and DELETED.* and stuff */
+#define MAX_MAILBOX_NAME 510
 #define MAX_MAILBOX_BUFFER 1024
 #define MAX_MAILBOX_PATH 4096
 
```

Comment:
The "enough space..." comment used to appear to describe MAX_MAILBOX_BUFFER, but now it appears to describe MAX_MAILBOX_NAME, and there's no longer an apparent explanation of why MAX_MAILBOX_BUFFER is twice the size.

The explanation in your previous comment makes sense, but a comment on a PR will be lost to history almost immediately...

Since these four defines are all very similarly named but different in intent, it'd be a lot clearer to future developers (or forgetful ones, like me) to just have a little comment block above them describing what they're for.

Maybe something like:

```
/* MAX_MAILBOX_CREATENAME - max length of a mailbox name during creation
 * MAX_MAILBOX_NAME - max length of a mailbox name, with space for delayed_delete munging
 * MAX_MAILBOX_BUFFER - buffer size that's large enough for all possible mailbox name operations
 * MAX_MAILBOX_PATH - max length of a mailbox's on disk path
 */
```

(hope I've got all those right, but if I didn't then that means the comment is even more important)

I wonder if we can define MAX_MAILBOX_NAME in terms of `(MAX_MAILBOX_CREATENAME + strlen(config_getstring(IMAPOPT_DELETEDPREFIX)) + expected_length_of_timestamp_suffix)` or something, instead of just hoping nobody configures a longer prefix than we anticipated.  I don't think it would be too expensive, libconfig tries quite hard to be very cheap to look up once loaded.  If we don't do this, we should probably update the deletedprefix entry in imapoptions to say that it shouldn't be longer than _n_.